src/Listener.php

```diff
@@ -59,9 +59,10 @@ abstract class Listener
      * Undocumented function
      *
      * @param Line $line
+     * @param Lexer $lexer
      * @return void
      */
-    abstract public function process(Line $line);
+    abstract public function process(Line $line, Lexer $lexer=null);
```

Comment:
@lode this would break current code and needs a major release.

Reply:
(also i assume $lexer could not be null, therefore it would be Lexer $lexer, but we can not introduce this in this PR as it will break all current custom listeners.)

Reply:
@lode the line object holds the lexer object already. So if there is really a need to access this in process method you can just do $line->lexer

Reply:
Yeah, I chose this as that object is protected. We would need to make it available via the line in that case. Shall I do that instead?